_posts/2021-06-23-devfile-v2-and-ide-plugins.md

```diff
@@ -0,0 +1,273 @@
+
+With devfile v2, there is still these kind of steps where a lot of yaml/json files are fetched, analyzed to result at the end in a set of DevWorkspace templates.
+
+The Che-Theia library allows to generate a single yaml file with all the analysis being done offline.
+
+npx @eclipse-che/che-theia-devworkspace-handler --devfile-url:[https://github.com/che-samples/spring-petclinic/tree/devfilev2](https://github.com/che-samples/spring-petclinic/tree/devfilev2) --output-file:$(pwd)/all-in-one.yaml
+
+Then this file can be used directly by `kubectl` :
+
+`kubectl apply -f all-in-one.yaml -n my-namespace`
+
+These templates can be stored in the devfile registry so Eclipse Che dashboard could directly use this content instead of processing at every click the full analysis.
+
+### Plug-in registry changes
+
+For the workspaces using devfile v1, Eclipse Che server fetches meta.yaml files from the plug-in registry. There are some limitations, for example defining plug-in preferences is only possible if there is a sidecar being defined, etc.
+
+With DevWorkspaces, the plug-in registry export now the content provided in different formats. It still export meta.yaml but it also expose devfile.yaml files for Eclipse Che editors definition and for some plug-ins that are not IDE plug-ins like che-machine-exec (library to be able connect to a specific container in a workspace/pod).
+
+Also the Che-Theia IDE plug-ins are now exposed by their che-theia-plugin.yaml fragment.
+
+Every Che-Theia plug-in fragment is generated from the [che-theia-plugins.yaml file](https://github.com/eclipse-che/che-plugin-registry/blob/main/che-theia-plugins.yaml).
+
+An hosted version of the che-plugin-registry is available after each commit at [https://eclipse-che.github.io/che-plugin-registry/main](https://eclipse-che.github.io/che-plugin-registry/main)
+
+Corresponding definition for`redhat/java` IDE plug-in is available at:
+
+[https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/redhat/java/latest/che-theia-plugin.yaml](https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/redhat/java/latest/che-theia-plugin.yaml)
+
+It references the dependencies of this IDE plug-in and the preferences (so .vsix are not grouped directly in the meta.yaml file)
+
+```yaml
+preferences:  
+  java.server.launchMode: Standard  
+dependencies:  
+  - vscjava/vscode-java-debug  
+  - vscjava/vscode-java-test
+```
+
+### Eclipse Che Devfile registry changes
+
+The index of the devfile registry is providing a list of devfiles to use. But now, with devfile V2 as some other optional files may be required, it’s better to reference a repository rather than a devfile.yaml
```

Comment:
```suggestion
The index of the Devfile registry is providing a list of Devfiles to use. But now, Devfile v2 as some other optional files may be required, it’s better to reference a repository rather than a single `devfile.yaml`.
```